FreeRTOS/FreeRTOS-Kernel
# Add Renesas RXv3 port layer supporting RXv3's double precision FPU

Description

As I told to Richard in the following thread, I'd like to post a pull request to add Renesas RXv3 port layer supporting RXv3's double precision FPU. I'll post one more pull request to add RTOS Demo using this port layer.

Any plan of Renesas RXv3 port layer supporting RXv3's double precision FPU?
[https://forums.freertos.org/t/any-plan-of-renesas-rxv3-port-layer-supporting-rxv3s-double-precision-fpu/9931](https://forums.freertos.org/t/any-plan-of-renesas-rxv3-port-layer-supporting-rxv3s-double-precision-fpu/9931)

This port layer is derived from Renesas RXv2 port layer by adding Cortex-A9's FPU support code of GCC port layer. Added code is not the same as Corteax-A9's FPU support code because CPU is not Cortex-A9 but RXv3 and more efficient code was discussed in the above thread.

This pull request includes additional changes from Renesas RXv2 port layer.

(1) In case of IAR port layer, ASM source was removed and inline asm code is added to C source because ASM source used another configuration file 'PriorityDefinitions.h' and having two configuration file (the other is 'FreeRTOSConfig.h') seems to be inconvenient for not only users but also code generation tool such as RX SmartConfigurator in the future. So, I (or Renesas) will post pull requests including the same modification for other port layers such as RX100, RX600, RX600v2  in the future.

(2) Synchronize three port.c of GNURX/CC-RX/ICCRX each other because comparison among three port.c becomes easier.

(3) Not only CC-RX port layer but also GNURX/ICCRX port layer use configINCLUDE_PLATFORM_H_INSTEAD_OF_IODEFINE_H because Renesas's recent code generation tool need the same modification to use 'platform.h'.

(4) Change code to avoid Improper access to an on-chip peripheral register. This was reported in the following thread.

Improper access to an on-chip peripheral register in Renesas RX port layer
[https://forums.freertos.org/t/improper-access-to-an-on-chip-peripheral-register-in-renesas-rx-port-layer/9851](https://forums.freertos.org/t/improper-access-to-an-on-chip-peripheral-register-in-renesas-rx-port-layer/9851)

(5) Add code to avoid compiler warning about unreferenced parameter in GNURX port layer..

(6) Add workaround for e2 studio CDT INDEXER (and maybe CODAN)'s warning.

(7) Fix typos such as mistaken comments of 'Accumulator 0' and 'Accumulator 1'.

Test Steps
-----------
(A) Build check is done with the following 2 * 3 * 3 = 18 cases using RTOS Demo program.

(A-1) Define/not define USE_FULL_REGISTER_INITIALISATION
(A-2) Define configUSE_TASK_DPFPU_SUPPORT = 1, 2, 0
(A-3) GNURX, CC-RX, ICCRX

(B) Execution check is done with the following ( 2 + 1 ) * 3 = 9 cases using RTOS Demo program.

(B-1) Define configUSE_TASK_DPFPU_SUPPORT = 1, 2 and expect NO RUNTIME ERROR (i.e. LED blinks with 3sec.)
(B-1') Define configUSE_TASK_DPFPU_SUPPORT = 0 and expect RUNTIME ERROR (i.e. LED blinks with 0.2sec.)
(B-2) GNURX, CC-RX, ICCRX

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
